mesonbuild/modules/gnome.py

```diff
@@ -1048,7 +1051,7 @@ def generate_gir(self, state: 'ModuleState', args: T.Tuple[T.List[T.Union[build.
         srcdir = os.path.join(state.environment.get_source_dir(), state.subdir)
         builddir = os.path.join(state.environment.get_build_dir(), state.subdir)
 
-        depends: T.List[T.Union['FileOrString', build.GeneratedTypes, build.Executable, build.SharedLibrary, build.StaticLibrary]] = []
+        depends: T.List[T.Union['FileOrString', 'build.GeneratedTypes', build.BuildTarget]] = []
```

Comment:
How did this ever work? Weird. build.GeneratedTypes is a typechecking-only variable but it never produced undefined variable errors when running live?

Reply:
It seems that the answer to this is: https://www.python.org/dev/peps/pep-0526/#runtime-effects-of-type-annotations

A variable inside a function doesn't get stored in `__annotations__` of the function, unlike class or module variables, and in fact isn't even evaluated...